src/pint/observatory/__init__.py

```diff
@@ -272,12 +295,19 @@ def get(cls, name):
 
         # Be case-insensitive
         name = name.lower()
-        # First see if name matches
+
+        # First see if name matches an already-registered observatory (or an alias)
         if name in cls._registry:
-            return cls._registry[name]
-        # Then look for aliases
-        if name in cls._alias_map:
-            return cls._registry[cls._alias_map[name]]
+            site = cls._registry[name]
+        elif name in cls._alias_map:
+            site = cls._registry[cls._alias_map[name]]
+        else:
+            site = None
+        if site is not None:
+            if overwrite and apply_gps2utc is not None:
+                # This will modify the Observatory object in the registry, so it will "stick" until overwritten
+                site.apply_gps2utc = apply_gps2utc
```

Comment:
Can we have a warning here telling the user to be extra careful?

Reply:
Added.